File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import java.util.List;
import java.util.Map;


import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojp.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;

@RestController
@RequestMapping("/brand")
public class BrandController {

	
	@Reference
	private BrandService brandService;


	@RequestMapping("/findAll.do")
	public List<TbBrand> findAll(){
		return brandService.findAll();
	}


	@RequestMapping("/findPage.do")
	public PageResult findPage(int page,int size){
		return brandService.findPage(page, size);
	}


	@RequestMapping("/add.do")
	public Result add(@RequestBody TbBrand brand){
		try {
			brandService.add(brand);
			return new Result(true,"新增成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false,"新增失败");
		}

	}

	@RequestMapping("/findOne.do")
	public TbBrand findOne(Long id){
		return brandService.findOne(id);
	}


	@RequestMapping("/update.do")
	public Result update(@RequestBody TbBrand brand){
		try {
			brandService.update(brand);
			return new Result(true,"修改成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false,"修改失败");
		}

	}



	@RequestMapping("/delete.do")
	public Result delete(Long[] ids){
		try {
			brandService.delete(ids);
			return new Result(true,"删除成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false,"删除失败");
		}

	}

	@RequestMapping("/search.do")
	public PageResult search(@RequestBody TbBrand brand,int page,int size){
			return  brandService.findPage(brand,page,size);
	}

	@RequestMapping("/selectOptionList.do")
	public List<Map> selectOptionList(){
		return brandService.selectOptionList();
	}

}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js

app.controller('brandController',function ($scope,$controller,brandService) {

 $controller('baseController',{$scope:$scope});


 $scope.findAll=function () {
brandService.findAll().success(
function (json) {
$scope.list=json;
}
);
}



$scope.findPage=function (page,size) {
brandService.findPage(page,size).success(
function (json) {
$scope.list=json.rows;
$scope.paginationConf.totalItems=json.totle;
}
)

};


$scope.add=function(){
var object=null;
if($scope.entity.id!=null){
object=brandService.update($scope.entity)
}else {
object=brandService.add($scope.entity)
}
object.success(
function (json) {
if(json.success){
$scope.reloadList();
}else {
alert(json.message);
}
}

)
}


$scope.findOne=function (id) {
brandService.findOne(id).success(
function (json) {
$scope.entity=json;
}

)
}




$scope.dele=function () {
brandService.dele($scope.selectIds).success(
function (json) {
if(json.success){
$scope.reloadList();
}else {
alert(json.message);
}
}

)

}

$scope.searchEntity={};

$scope.search=function (page,size) {
brandService.search(page,size,$scope.searchEntity).success(
function (json) {
$scope.list=json.rows;
$scope.paginationConf.totalItems=json.totle;
}
)
}

})